snooty/test_parser.py

```diff
@@ -612,6 +612,92 @@ def test_accidental_indentation() -> None:
     assert len(diagnostics) == 1
 
 
+def test_glossary_node() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(ROOT_PATH, "", source="./")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+.. glossary::
+  :sorted:
+
+  _id
+    foofoobarbar
+
+  index 
+    foofoofoofoobarbarbarbar
+
+  $cmd
+    foobar
+    
+  aggregate
+    foofoofoobarbarbar
+  
+  
+""",
+    )
+    page.finish(diagnostics)
+
+    check_ast_testing_string(
+        page.ast,
+        """
+<root> 
+  <directive name="glossary" sorted="True">
+    <definitionList>       
+      <definitionListItem>         
+        <term>           
+          <text>$cmd</text>
+          <inline_target domain="std" name="term">            
+            <target_identifier ids="['cmd']">
```

Comment:
I know in the jira ticket you had the id as `$cmd` but I looked into docutils make_id and the docs say, "ID and NAME tokens must begin with a letter ([A-Za-z]) and may be followed by any number of letters, digits ([0-9]), hyphens (“-”), underscores (“_”), colons (”:”), and periods (”.”)." ([source](ID and NAME tokens must begin with a letter ([A-Za-z]) and may be followed by any number of letters, digits ([0-9]), hyphens (“-”), underscores (“_”), colons (”:”), and periods (”.”).)